lib/concurrent/edge/future.rb

```diff
@@ -100,25 +128,26 @@ def initialize(promise, default_executor = :io)
         ensure_ivar_visibility!
       end
 
+      # @return [:pending, :completed]
       def state
-        @State.get
+        @State.get.to_sym
```

Comment:
In this case it cannot. `@State.get` is set only internally and it can be only Symbol or special private `Success`of `Failed` objects which are overriding `to_sym` to return `:success` or `:failed` symbols.